### DISCUSSION TITLE
STM32F103 I2C stuck

### DISCUSSION THREAD

**Discussion by arozhkov2001** [stm32duino/Arduino_Core_STM32#2232] [Q&A] [answered]
Hi, im trying to write code to use stm32f103c8t6 and bq76940, i tried two different libraries for bq76940 and same effect, im tried lower i2c speed but without changes, on atmel chips working without any problems. Stm32 stuck totally because blink doest work too after chip freeze
[i2c_scanner.zip](https://github.com/stm32duino/Arduino_Core_STM32/files/13766896/i2c_scanner.zip)

**Comment by fpistm** [comment]
Hi @arozhkov2001 
Do you have a Pull-up resistor on each I2C line?

**Comment by arozhkov2001** [comment]
Yes, Of course

**Reply by fpistm** [reply]
Please try the standard I2C scanner.

**Reply by arozhkov2001** [reply]
Standart i2c scanner working

**Reply by fpistm** [reply]
So it is link to another part of your code. I bet the one for bq76940. Ask support to the library owner.

**Reply by arozhkov2001** [reply]
Now im using Stm32f072cbt6 with this code without changes and all work without problem, so i think problem with stm32f103c8t6 (also im tried in stm32f103rct6(discovery) and same problem) because for bq76940 im founded only 2 libraries and they both are stuck,

**Reply by fpistm** [reply, marked as answer]
OK. one other possibilities is that you uses a pin which remove the AF of the I2C pins you used. STM32F1 is limited on the AF side.
